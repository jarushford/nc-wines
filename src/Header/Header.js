import React from 'react'
import '../main.scss'

export default function Header() {
  return (
    <header className="header">
      <h1>North Carolina Wines</h1>
      <button
        onClick={() => window.scrollTo({ top: 500, behavior: 'smooth' })}
      >
        Test Endpoints
      </button>
      <button
        onClick={() => window.scrollTo({ top: 1900, behavior: 'smooth' })}
      >
        Available Endpoints
      </button>
    </header>
  )
}