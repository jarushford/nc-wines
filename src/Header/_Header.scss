.header {
  box-sizing: border-box;
  background: url('./assets/nc-winery-1500x609.jpg');
  background-size: cover;
  height: 700px;
  background-position: center;
  background-color: rgba(255, 255, 255, 0.4);
  background-blend-mode: lighten;
  box-shadow: inset 0px -40px 40px 10px rgba(255, 255, 255, 1);
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
  margin-bottom: 60px;

  button { @include btn; }

  h1 {
    font-family: 'Satisfy', cursive;
    font-size: 3.7rem;
    margin: -30px auto 15px
  }
}