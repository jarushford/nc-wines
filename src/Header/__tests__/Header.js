import React from 'react'
import { shallow } from 'enzyme'
import Header from '../Header'

describe('Header', () => {
  it('Sould match the snapshot', () => {
    const wrapper = shallow(<Header />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should scroll on first button click', () => {
    const wrapper = shallow(<Header />)
    window.scrollTo = jest.fn().mockImplementation(() => true)
    wrapper.find('button').first().simulate('click')
    expect(wrapper).toMatchSnapshot()
  })

  it('should scroll on second button click', () => {
    const wrapper = shallow(<Header />)
    window.scrollTo = jest.fn().mockImplementation(() => true)
    wrapper.find('button').last().simulate('click')
    expect(wrapper).toMatchSnapshot()
  })
})