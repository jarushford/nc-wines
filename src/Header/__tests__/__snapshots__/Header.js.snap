// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Sould match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          North Carolina Wines
        </h1>,
        <button
          onClick={[Function]}
        >
          Test Endpoints
        </button>,
        <button
          onClick={[Function]}
        >
          Available Endpoints
        </button>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "North Carolina Wines",
        },
        "ref": null,
        "rendered": "North Carolina Wines",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Test Endpoints",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Test Endpoints",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Available Endpoints",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Available Endpoints",
        "type": "button",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            North Carolina Wines
          </h1>,
          <button
            onClick={[Function]}
          >
            Test Endpoints
          </button>,
          <button
            onClick={[Function]}
          >
            Available Endpoints
          </button>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "North Carolina Wines",
          },
          "ref": null,
          "rendered": "North Carolina Wines",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Test Endpoints",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Test Endpoints",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Available Endpoints",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Available Endpoints",
          "type": "button",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Header should scroll on first button click 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          North Carolina Wines
        </h1>,
        <button
          onClick={[Function]}
        >
          Test Endpoints
        </button>,
        <button
          onClick={[Function]}
        >
          Available Endpoints
        </button>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "North Carolina Wines",
        },
        "ref": null,
        "rendered": "North Carolina Wines",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Test Endpoints",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Test Endpoints",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Available Endpoints",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Available Endpoints",
        "type": "button",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            North Carolina Wines
          </h1>,
          <button
            onClick={[Function]}
          >
            Test Endpoints
          </button>,
          <button
            onClick={[Function]}
          >
            Available Endpoints
          </button>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "North Carolina Wines",
          },
          "ref": null,
          "rendered": "North Carolina Wines",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Test Endpoints",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Test Endpoints",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Available Endpoints",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Available Endpoints",
          "type": "button",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Header should scroll on second button click 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          North Carolina Wines
        </h1>,
        <button
          onClick={[Function]}
        >
          Test Endpoints
        </button>,
        <button
          onClick={[Function]}
        >
          Available Endpoints
        </button>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "North Carolina Wines",
        },
        "ref": null,
        "rendered": "North Carolina Wines",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Test Endpoints",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Test Endpoints",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Available Endpoints",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Available Endpoints",
        "type": "button",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            North Carolina Wines
          </h1>,
          <button
            onClick={[Function]}
          >
            Test Endpoints
          </button>,
          <button
            onClick={[Function]}
          >
            Available Endpoints
          </button>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "North Carolina Wines",
          },
          "ref": null,
          "rendered": "North Carolina Wines",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Test Endpoints",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Test Endpoints",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Available Endpoints",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Available Endpoints",
          "type": "button",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
